src/api/app/models/search_finder.rb

```diff
@@ -0,0 +1,89 @@
+class SearchFinder
+  include ActiveModel::Validations
+
+  attr_reader :included_classes, :relation, :what, :render_all, :params, :search_items, :preloaded_classes
+
+  validates :what, inclusion: [:package, :project, :repository, :request, :person, :channel, :channel_binary, :released_binary, :issue]
+
+  def initialize(what:, search_items: [], render_all: false, params: {})
+    @what = what
+    @render_all = render_all
+    @search_items = search_items
+    @params = params
+    @included_classes = []
+    @preloaded_classes = []
+    @relation = nil
```

Comment:
This line isn't needed because an uninitialized instance variable is already going to evaluate to `nil`.

```suggestion
```